streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -248,21 +250,33 @@ public static void verify(final String kafka, final boolean withRepartitioning)
         System.out.flush();
     }
 
-    private static void ensureStreamsApplicationDown(final Admin adminClient) {
-
+    private static void ensureStreamsApplicationDown(final Admin adminClient, final String groupProtocol) {
         final long maxWaitTime = System.currentTimeMillis() + MAX_IDLE_TIME_MS;
-        ConsumerGroupDescription description;
-        do {
-            description = getConsumerGroupDescription(adminClient);
-
-            if (System.currentTimeMillis() > maxWaitTime && !description.members().isEmpty()) {
-                throw new RuntimeException(
-                    "Streams application not down after " + (MAX_IDLE_TIME_MS / 1000L) + " seconds. " +
-                        "Group: " + description
-                );
-            }
-            sleep(1000L);
-        } while (!description.members().isEmpty());
+        if (Objects.equals(groupProtocol, "streams")) {
+            StreamsGroupDescription description;
+            do {
+                description = getStreamsGroupDescription(adminClient);
+                if (System.currentTimeMillis() > maxWaitTime && !description.members().isEmpty()) {
+                    throw new RuntimeException(
+                        "Streams application not down after " + MAX_IDLE_TIME_MS / 1000L + " seconds. " +
+                            "Group: " + description
+                    );
+                }
+                sleep(1000L);
+            } while (!description.members().isEmpty());
+        } else {
+            ConsumerGroupDescription description;
+            do {
+                description = getConsumerGroupDescription(adminClient);
+                if (System.currentTimeMillis() > maxWaitTime && !description.members().isEmpty()) {
+                    throw new RuntimeException(
+                        "Streams application not down after " + MAX_IDLE_TIME_MS / 1000L + " seconds. " +
+                            "Group: " + description
+                    );
+                }
+                sleep(1000L);
+            } while (!description.members().isEmpty());
+        }
```

Comment:
nit: I think this could be refactored into one `do-while` loop with the `if-else` separating out the `StreamsGroupDescription` vs. the `ConsumerGroupDescription`

Reply:
A bit of an awkward refactoring, because `description` is used in the while condition, and the two description types only share `Object` as common ancestor. Tried my best